Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -97,6 +97,17 @@ s32 WiiSockMan::GetNetErrorCode(s32 ret, std::string_view caller, bool is_rw)
 {
 #ifdef _WIN32
   s32 error_code = WSAGetLastError();
+  // Some programs might hijack WinSock2 (e.g. ReShade) and alter the expected return value.
+  if (error_code == WSAEINVAL && caller.compare("SO_CONNECT") == 0)
```

Comment:
```suggestion
  if (error_code == WSAEINVAL && caller == "SO_CONNECT")
```

Also, should you also add `WSAEWOULDBLOCK` here? 
[The documentation you linked](https://learn.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-connect) says that both `WSAEINVAL` and `WSAEWOULDBLOCK` should be handled the same as `WSAEALREADY`.

Reply:
This won't be necessary as `WSAEWOULDBLOCK` is properly handled by `TranslateErrorCode`. `WSAEINVAL` isn't and returns a default error code. That shouldn't be a problem in normal circumstances except for `connect` on Windows as it shouldn't be considered as a failure.